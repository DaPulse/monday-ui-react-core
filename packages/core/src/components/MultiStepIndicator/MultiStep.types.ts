export type MultiStepType = "primary" | "success" | "danger" | "dark";

export type StepStatus = "pending" | "active" | "fulfilled";

export type TextPlacement = "horizontal" | "vertical";

export type MultiStepSize = "regular" | "compact";
