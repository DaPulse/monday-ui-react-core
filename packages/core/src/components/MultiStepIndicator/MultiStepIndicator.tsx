import cx from "classnames";
import React, { forwardRef, useCallback, useMemo } from "react";
import Check from "../../components/Icon/Icons/components/Check";
import Divider from "../../components/Divider/Divider";
import { NOOP } from "../../utils/function-utils";
import StepIndicator from "./components/StepIndicator/StepIndicator";
import {
  MultiStepType as MultiStepTypeEnum,
  Size as SizeEnum,
  StepStatus as StepStatusEnum,
  TextPlacement as TextPlacementEnum
} from "./MultiStepConstants";
import { MultiStepType, MultiStepSize, StepStatus, TextPlacement } from "./MultiStep.types";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../tests/constants";
import { SubIcon, VibeComponent, VibeComponentProps, withStaticProps } from "../../types";
import styles from "./MultiStepIndicator.module.scss";

export type Step = {
  titleText: string;
  subtitleText: string;
  status: StepStatus;
};

export interface MultiStepIndicatorProps extends VibeComponentProps {
  steps?: Step[];
  type?: MultiStepType;
  stepComponentClassName?: string;
  dividerComponentClassName?: string;
  fulfilledStepIcon?: SubIcon;
  fulfilledStepIconType?: "svg" | "font";
  isFulfilledStepDisplayNumber?: boolean;
  onClick?: (stepNumber: number) => void;
  textPlacement?: TextPlacement;
  size?: MultiStepSize;
}

const MultiStepIndicator: VibeComponent<MultiStepIndicatorProps, HTMLOListElement> & {
  types?: typeof MultiStepTypeEnum;
  stepStatuses?: typeof StepStatusEnum;
  textPlacements?: typeof TextPlacementEnum;
  sizes?: typeof SizeEnum;
} = forwardRef(
  (
    {
      className,
      steps = [],
      type = "primary",
      stepComponentClassName,
      dividerComponentClassName,
      fulfilledStepIcon = Check,
      fulfilledStepIconType = "svg",
      isFulfilledStepDisplayNumber = false,
      onClick = NOOP,
      textPlacement = "horizontal",
      id,
      size,
      "data-testid": dataTestId
    }: MultiStepIndicatorProps,
    ref
  ) => {
    const finalSize = textPlacement === "vertical" ? "regular" : size;

    const renderHorizontalStepIndicator = useCallback(
      (step: Step, index: number) => {
        return (
          <React.Fragment key={`${step.titleText}_${index}`}>
            <StepIndicator
              {...step}
              stepNumber={index + 1}
              type={type}
              stepComponentClassName={stepComponentClassName}
              fulfilledStepIcon={fulfilledStepIcon}
              fulfilledStepIconType={fulfilledStepIconType}
              onClick={onClick}
              isFulfilledStepDisplayNumber={isFulfilledStepDisplayNumber}
              size={finalSize}
            />
            {index !== steps.length - 1 && (
              <Divider
                className={cx(styles.divider, dividerComponentClassName, {
                  [styles.compact]: finalSize === "compact"
                })}
              />
            )}
          </React.Fragment>
        );
      },
      [
        onClick,
        isFulfilledStepDisplayNumber,
        type,
        stepComponentClassName,
        fulfilledStepIcon,
        fulfilledStepIconType,
        dividerComponentClassName,
        steps.length,
        finalSize
      ]
    );

    const renderVerticalStepIndicator = useCallback(
      (step: Step, index: number) => {
        return (
          <StepIndicator
            {...step}
            key={`${step.titleText}_${index}`}
            stepNumber={index + 1}
            type={type}
            stepComponentClassName={stepComponentClassName}
            fulfilledStepIcon={fulfilledStepIcon}
            fulfilledStepIconType={fulfilledStepIconType}
            onClick={onClick}
            isFollowedByDivider={index !== steps.length - 1}
            stepDividerClassName={cx(styles.divider, dividerComponentClassName)}
            isVertical
            isFulfilledStepDisplayNumber={isFulfilledStepDisplayNumber}
          />
        );
      },
      [
        onClick,
        isFulfilledStepDisplayNumber,
        type,
        stepComponentClassName,
        fulfilledStepIcon,
        fulfilledStepIconType,
        dividerComponentClassName,
        steps.length
      ]
    );

    const stepRenderer = useMemo(
      () => (textPlacement === "vertical" ? renderVerticalStepIndicator : renderHorizontalStepIndicator),
      [textPlacement, renderVerticalStepIndicator, renderHorizontalStepIndicator]
    );

    return (
      <ol
        ref={ref}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MULTI_STEP_INDICATOR, id)}
        className={cx(styles.wrapper, className)}
      >
        {steps.map(stepRenderer)}
      </ol>
    );
  }
);

export default withStaticProps(MultiStepIndicator, {
  types: MultiStepTypeEnum,
  stepStatuses: StepStatusEnum,
  textPlacements: TextPlacementEnum,
  sizes: SizeEnum
});
