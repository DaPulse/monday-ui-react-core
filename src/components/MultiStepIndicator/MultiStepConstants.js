export const MULTI_STEP_TYPES = {
    PRIMARY: "primary",
    SUCCESS: "success",
    DANGER: "danger",
    DARK: "dark"
}