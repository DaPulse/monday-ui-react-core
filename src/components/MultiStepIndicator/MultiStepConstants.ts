export enum MULTI_STEP_TYPES {
  PRIMARY = "primary",
  SUCCESS = "success",
  DANGER = "danger",
  DARK = "dark"
}

export enum STEP_STATUSES {
  PENDING = "pending",
  ACTIVE = "active",
  FULFILLED = "fulfilled"
}

export enum TEXT_PLACEMENTS {
  HORIZONTAL = "horizontal",
  VERTICAL = "vertical"
}

export enum SIZES {
  REGULAR = "regular",
  COMPACT = "compact"
}
