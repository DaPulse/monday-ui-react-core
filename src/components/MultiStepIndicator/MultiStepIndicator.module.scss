@import "../../styles/themes";

.wrapper {
  display: flex;
}

.divider {
  @include theme-prop(background-color, ui-border-color);
  margin: 16px;
  height: 1px;
  width: 48px;
}
