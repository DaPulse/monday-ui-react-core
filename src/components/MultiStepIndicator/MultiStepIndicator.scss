@import "../../styles/themes.scss";
@import "../../styles/typography.scss";

.multi-step-indicator--wrapper {
  display: flex;
}
