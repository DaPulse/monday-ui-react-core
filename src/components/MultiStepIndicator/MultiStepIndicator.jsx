import React, { useRef, forwardRef } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import useMergeRefs from "../../hooks/useMergeRefs";
import Icon from "../Icon/Icon";
import Check from "../Icon/Icons/components/Check";
import Divider from "../Divider/Divider";
import StepIndicator from "./components/StepIndicator/StepIndicator";
import { MULTI_STEP_TYPES, STEP_STATUSES } from "./MultiStepConstants";
import "./MultiStepIndicator.scss";

const MultiStepIndicator = forwardRef(
  (
    {
      className,
      type,
      steps,
      stepComponentClassName,
      dividerComponentClassName,
      fulfilledStepIcon,
      fulfilledStepIconType
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    const baseClassName = "multi-step-indicator--wrapper";
    const defaultDividerClassName = `${baseClassName}__divider`;
    return (
      <div ref={mergedRef} className={cx(baseClassName, className)}>
        {steps.map((step, index) => {
          return (
            <>
              <StepIndicator
                {...step}
                stepNumber={index + 1}
                type={type}
                stepComponentClassName={stepComponentClassName}
                fulfilledStepIcon={fulfilledStepIcon}
                fulfilledStepIconType={fulfilledStepIconType}
              />
              {index !== steps.length - 1 && (
                <Divider classname={cx(defaultDividerClassName, dividerComponentClassName)} />
              )}
            </>
          );
        })}
      </div>
    );
  }
);

MultiStepIndicator.types = MULTI_STEP_TYPES;
MultiStepIndicator.stepStatuses = STEP_STATUSES;

MultiStepIndicator.propTypes = {
  /** For overriding the container class styles. */
  className: PropTypes.string,
  type: PropTypes.oneOf([
    MultiStepIndicator.types.PRIMARY,
    MultiStepIndicator.types.SUCCESS,
    MultiStepIndicator.types.DANGER,
    MultiStepIndicator.types.DARK
  ]),
  /** Array of objects of the specified format. */
  steps: PropTypes.arrayOf({
    titleText: PropTypes.string,
    subtitleText: PropTypes.string,
    status: PropTypes.oneOf([
      MultiStepIndicator.stepStatuses.PENDING,
      MultiStepIndicator.stepStatuses.ACTIVE,
      MultiStepIndicator.stepStatuses.FULFILLED
    ])
  }).isRequired,
  /** For overriding the styles of the step component - container of number/check and texts. */
  stepComponentClassName: PropTypes.string,
  /** For overriding the step-dividers styles. */
  dividerComponentClassName: PropTypes.string,
  /** For overriding the 'fulfilled' step's icon. Is passed directly to an Icon component. */
  fulfilledStepIcon: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  /** For overriding the 'fulfilled' step's icon type. Necessary when passing a string in the "fulfilledStepIcon" prop. */
  fulfilledStepIconType: PropTypes.oneOf([Icon.type.SVG, Icon.type.ICON_FONT])
};

MultiStepIndicator.defaultProps = {
  className: "",
  stepComponentClassName: "",
  dividerComponentClassName: "",
  type: MultiStepIndicator.types.PRIMARY,
  steps: [],
  fulfilledStepIcon: Check,
  fulfilledStepIconType: Icon.type.SVG
};

export default MultiStepIndicator;
