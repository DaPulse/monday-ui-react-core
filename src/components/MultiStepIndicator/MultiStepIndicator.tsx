import cx from "classnames";
import React, { useRef, forwardRef, useMemo, useCallback } from "react";
import useMergeRefs from "../../hooks/useMergeRefs";
import Check from "../../components/Icon/Icons/components/Check";
import Divider from "../../components/Divider/Divider";
import { NOOP } from "../../utils/function-utils";
import StepIndicator from "./components/StepIndicator/StepIndicator";
import { SIZES, MULTI_STEP_TYPES, STEP_STATUSES, TEXT_PLACEMENTS } from "./MultiStepConstants";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../tests/constants";
import styles from "./MultiStepIndicator.module.scss";
import { withStaticProps, VibeComponentProps, VibeComponent, SubIcon } from "../../types";
import { IconType } from "../Icon/IconConstants";

export type Step = {
  titleText: string;
  subtitleText: string;
  status: STEP_STATUSES;
};

interface MultiStepIndicatorProps extends VibeComponentProps {
  steps: Step[];
  type?: MULTI_STEP_TYPES;
  stepComponentClassName?: string;
  dividerComponentClassName?: string;
  fulfilledStepIcon?: SubIcon;
  fulfilledStepIconType?: IconType.SVG | IconType.ICON_FONT;
  isFulfilledStepDisplayNumber?: boolean;
  onClick?: (stepNumber: number) => void;
  textPlacement?: TEXT_PLACEMENTS;
  size?: SIZES;
}

const MultiStepIndicator: VibeComponent<MultiStepIndicatorProps> & {
  types?: typeof MULTI_STEP_TYPES;
  stepStatuses?: typeof STEP_STATUSES;
  textPlacements?: typeof TEXT_PLACEMENTS;
  sizes?: typeof SIZES;
} = forwardRef(
  (
    {
      className,
      steps,
      type = MULTI_STEP_TYPES.PRIMARY,
      stepComponentClassName,
      dividerComponentClassName,
      fulfilledStepIcon = Check,
      fulfilledStepIconType = IconType.SVG,
      isFulfilledStepDisplayNumber = false,
      onClick = NOOP,
      textPlacement = TEXT_PLACEMENTS.HORIZONTAL,
      id,
      size,
      "data-testid": dataTestId
    },
    ref
  ) => {
    const componentRef = useRef<HTMLInputElement>(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    const finalSize = textPlacement === TEXT_PLACEMENTS.VERTICAL ? SIZES.REGULAR : size;

    const renderHorizontalStepIndicator = useCallback(
      (step: Step, index: number) => {
        return (
          <React.Fragment key={`${step.titleText}_${index}`}>
            <StepIndicator
              {...step}
              stepNumber={index + 1}
              type={type}
              stepComponentClassName={stepComponentClassName}
              fulfilledStepIcon={fulfilledStepIcon}
              fulfilledStepIconType={fulfilledStepIconType}
              onClick={onClick}
              isFulfilledStepDisplayNumber={isFulfilledStepDisplayNumber}
              size={finalSize}
            />
            {index !== steps.length - 1 && (
              <Divider
                classname={cx(styles.divider, dividerComponentClassName, {
                  [styles.compact]: finalSize === SIZES.COMPACT
                })}
              />
            )}
          </React.Fragment>
        );
      },
      [
        onClick,
        isFulfilledStepDisplayNumber,
        type,
        stepComponentClassName,
        fulfilledStepIcon,
        fulfilledStepIconType,
        dividerComponentClassName,
        steps.length,
        finalSize
      ]
    );

    const renderVerticalStepIndicator = useCallback(
      (step: Step, index: number) => {
        return (
          <StepIndicator
            {...step}
            key={`${step.titleText}_${index}`}
            stepNumber={index + 1}
            type={type}
            stepComponentClassName={stepComponentClassName}
            fulfilledStepIcon={fulfilledStepIcon}
            fulfilledStepIconType={fulfilledStepIconType}
            onClick={onClick}
            isFollowedByDivider={index !== steps.length - 1}
            stepDividerClassName={cx(styles.divider, dividerComponentClassName)}
            isVertical
            isFulfilledStepDisplayNumber={isFulfilledStepDisplayNumber}
          />
        );
      },
      [
        onClick,
        isFulfilledStepDisplayNumber,
        type,
        stepComponentClassName,
        fulfilledStepIcon,
        fulfilledStepIconType,
        dividerComponentClassName,
        steps.length
      ]
    );

    const stepRenderer = useMemo(
      () => (textPlacement === TEXT_PLACEMENTS.VERTICAL ? renderVerticalStepIndicator : renderHorizontalStepIndicator),
      [textPlacement, renderVerticalStepIndicator, renderHorizontalStepIndicator]
    );

    return (
      <ol
        ref={mergedRef}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MULTI_STEP_INDICATOR, id)}
        className={cx(styles.wrapper, className)}
      >
        {steps.map(stepRenderer)}
      </ol>
    );
  }
);

export default withStaticProps(MultiStepIndicator, {
  types: MULTI_STEP_TYPES,
  stepStatuses: STEP_STATUSES,
  textPlacements: TEXT_PLACEMENTS,
  sizes: SIZES
});
