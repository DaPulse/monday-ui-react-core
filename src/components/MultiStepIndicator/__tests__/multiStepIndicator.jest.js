import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import MultiStepIndicator from "../MultiStepIndicator";

describe("MultiStepIndicator", () => {
  describe("Render tests", () => {
    it("Renders correctly with empty props", () => {
      const tree = renderer.create(<MultiStepIndicator />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("Renders correctly with non-default props", () => {
      const exampleSteps = [
        {
          status: MultiStepIndicator.stepStatuses.FULFILLED,
          titleText: "Title",
          subtitleText: "Subtitle"
        },
        {
          status: MultiStepIndicator.stepStatuses.ACTIVE,
          titleText: "Active",
          subtitleText: "Active Subtitle"
        },
        {
          status: MultiStepIndicator.stepStatuses.PENDING,
          titleText: "Pending",
          subtitleText: "Pending Subtitle"
        }
      ];
      const tree = renderer.create(<MultiStepIndicator type={MultiStepIndicator.types.SUCCESS} steps={exampleSteps} />);
      expect(tree).toMatchSnapshot();
    });

    it("Render correctly with clickable steps", () => {
      const exampleSteps = [
        {
          status: MultiStepIndicator.stepStatuses.FULFILLED,
          titleText: "Title",
          subtitleText: "Subtitle"
        },
        {
          status: MultiStepIndicator.stepStatuses.ACTIVE,
          titleText: "Active",
          subtitleText: "Active Subtitle"
        },
        {
          status: MultiStepIndicator.stepStatuses.PENDING,
          titleText: "Pending",
          subtitleText: "Pending Subtitle"
        }
      ];

      const callback = () => {
        console.log("test");
      };

      const tree = renderer.create(
        <MultiStepIndicator type={MultiStepIndicator.types.SUCCESS} steps={exampleSteps} onClick={callback} />
      );
      expect(tree).toMatchSnapshot();
    });

    it("onClick works and is called once", () => {
      const exampleSteps = [
        {
          status: MultiStepIndicator.stepStatuses.FULFILLED,
          titleText: "Title",
          subtitleText: "Subtitle"
        },
        {
          status: MultiStepIndicator.stepStatuses.ACTIVE,
          titleText: "Active",
          subtitleText: "Active Subtitle"
        }
      ];

      const stepClickMock = jest.fn();

      const multiStepIndicatorComponent = render(
        <MultiStepIndicator type={MultiStepIndicator.types.SUCCESS} steps={exampleSteps} onClick={stepClickMock} />
      );

      const step = multiStepIndicatorComponent.getByText("Title");

      act(() => {
        fireEvent.mouseOver(step);
        jest.advanceTimersByTime(1000);
        fireEvent.click(step);
      });

      jest.advanceTimersByTime(1000);
      expect(stepClickMock.mock.calls.length).toBe(1);
    });
  });
});
