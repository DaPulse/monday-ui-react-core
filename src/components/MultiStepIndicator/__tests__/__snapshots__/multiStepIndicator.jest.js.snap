// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiStepIndicator Render tests Render correctly with clickable steps 1`] = `
<ol
  className="multi-step-indicator--wrapper"
>
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-fulfilled clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-fulfilled__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-fulfilled__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-fulfilled__number-container__text"
        >
          <svg
            aria-hidden={true}
            className="icon_component monday-style-step-indicator-component__number-container__text__check-icon icon_component--no-focus-style"
            fill="currentColor"
            height="16"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="16"
          >
            <path
              clipRule="evenodd"
              d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-fulfilled__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-fulfilled__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          fulfilled
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-fulfilled__text-container__title__text"
        >
          Title
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-fulfilled__text-container__subtitle__text"
      >
        Subtitle
      </span>
    </div>
  </li>
  <div
    className="monday-style-divider multi-step-indicator--wrapper__divider monday-style-divider--horizontal"
  />
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-active clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-active__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-active__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-active__number-container__text"
        >
          2
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-active__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-active__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          active
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-active__text-container__title__text"
        >
          Active
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-active__text-container__subtitle__text"
      >
        Active Subtitle
      </span>
    </div>
  </li>
  <div
    className="monday-style-divider multi-step-indicator--wrapper__divider monday-style-divider--horizontal"
  />
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-pending clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-pending__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-pending__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-pending__number-container__text"
        >
          3
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-pending__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-pending__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          pending
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-pending__text-container__title__text"
        >
          Pending
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-pending__text-container__subtitle__text"
      >
        Pending Subtitle
      </span>
    </div>
  </li>
</ol>
`;

exports[`MultiStepIndicator Render tests Renders correctly with empty props 1`] = `
<ol
  className="multi-step-indicator--wrapper"
/>
`;

exports[`MultiStepIndicator Render tests Renders correctly with non-default props 1`] = `
<ol
  className="multi-step-indicator--wrapper"
>
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-fulfilled clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-fulfilled__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-fulfilled__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-fulfilled__number-container__text"
        >
          <svg
            aria-hidden={true}
            className="icon_component monday-style-step-indicator-component__number-container__text__check-icon icon_component--no-focus-style"
            fill="currentColor"
            height="16"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="16"
          >
            <path
              clipRule="evenodd"
              d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-fulfilled__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-fulfilled__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          fulfilled
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-fulfilled__text-container__title__text"
        >
          Title
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-fulfilled__text-container__subtitle__text"
      >
        Subtitle
      </span>
    </div>
  </li>
  <div
    className="monday-style-divider multi-step-indicator--wrapper__divider monday-style-divider--horizontal"
  />
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-active clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-active__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-active__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-active__number-container__text"
        >
          2
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-active__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-active__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          active
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-active__text-container__title__text"
        >
          Active
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-active__text-container__subtitle__text"
      >
        Active Subtitle
      </span>
    </div>
  </li>
  <div
    className="monday-style-divider multi-step-indicator--wrapper__divider monday-style-divider--horizontal"
  />
  <li
    className="monday-style-clickable monday-style-step-indicator-component monday-style-step-indicator-component--type-success monday-style-step-indicator-component--status-pending clickable monday-style-clickable--disable-text-selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div
      className="monday-style-step-indicator-component__number-divider-container monday-style-step-indicator-component--type-success__number-divider-container monday-style-step-indicator-component--status-pending__number-divider-container"
    >
      <div
        className="monday-style-step-indicator-component__number-container monday-style-step-indicator-component--type-success__number-container monday-style-step-indicator-component--status-pending__number-container"
        role="button"
        tabIndex="0"
      >
        <span
          className="monday-style-step-indicator-component__number-container__text monday-style-step-indicator-component--type-success__number-container__text monday-style-step-indicator-component--status-pending__number-container__text"
        >
          3
        </span>
      </div>
    </div>
    <div
      className="monday-style-step-indicator-component__text-container monday-style-step-indicator-component--type-success__text-container monday-style-step-indicator-component--status-pending__text-container"
    >
      <div
        className="monday-style-step-indicator-component__text-container__title monday-style-step-indicator-component--type-success__text-container__title monday-style-step-indicator-component--status-pending__text-container__title"
      >
        <span
          className="hidden-text--wrapper"
          id="hiddenText"
        >
          pending
        </span>
         
        <span
          className="monday-style-step-indicator-component__text-container__title__text monday-style-step-indicator-component--type-success__text-container__title__text monday-style-step-indicator-component--status-pending__text-container__title__text"
        >
          Pending
        </span>
      </div>
      <span
        className="monday-style-step-indicator-component__text-container__subtitle__text monday-style-step-indicator-component--type-success__text-container__subtitle__text monday-style-step-indicator-component--status-pending__text-container__subtitle__text"
      >
        Pending Subtitle
      </span>
    </div>
  </li>
</ol>
`;
